akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -38,6 +39,7 @@ object ShardCoordinator {
    * INTERNAL API
    * Factory method for the [[akka.actor.Props]] of the [[ShardCoordinator]] actor.
    */
+  @silent("deprecated")
```

Comment:
Same as above, no user will ever call this and see the deprecation

Reply:
`class PersistentShardCoordinator` is not private[akka] and marked deprecated, and this private function calls that. Thus the `@silent` here.